src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -317,18 +319,31 @@ private void ChooseActions(in Entity entity, ref MicrobeAI ai, ref CompoundAbsor
             return;
         }
 
+        float atpLevel = compounds.GetCompoundAmount(Compound.ATP);
+
         // If this microbe is out of ATP, pick an amount of time to rest
-        if (compounds.GetCompoundAmount(Compound.ATP) < 1.0f)
+        if (atpLevel < 1.0f)
         {
             // Keep the maximum at 95% full, as there is flickering when near full
             ai.ATPThreshold = 0.95f * speciesFocus / Constants.MAX_SPECIES_FOCUS;
         }
 
         if (ai.ATPThreshold > MathUtils.EPSILON)
         {
-            if (compounds.GetCompoundAmount(Compound.ATP) <
-                compounds.GetCapacityForCompound(Compound.ATP) * ai.ATPThreshold)
+            if (atpLevel < compounds.GetCapacityForCompound(Compound.ATP) * ai.ATPThreshold)
             {
+                if (cellHealth.CurrentHealth > 2 * Constants.ENGULF_NO_ATP_DAMAGE)
+                {
+                    // even if we are out of ATP and there is microbe nearby, engulf them.
+                    // make sure engulfing doesn't kill the cell
+                    bool isMicrobeHunting = CheckForHuntingConditions(ref ai, ref position, ref organelles,
```

Comment:
I would have asked to add a blank line for readability before `if (isMicrobeHunting)` but I think it makes more sense to just inline this method call in there.